samplesheets/views_api.py

```diff
@@ -310,6 +323,240 @@ def post(self, request, *args, **kwargs):
+    """
+
+    http_method_names = ['get']
+    permission_required = 'samplesheets.edit_sheet'
+
+    def get(self, request, *args, **kwargs):
+        """GET request for listing iRODS access tickets"""
+        project = self.get_project()
+        tickets = IrodsAccessTicket.objects.filter(
+            study__investigation__project=project
+        )
+        active = request.query_params.get('active', '0')
+        active = bool(int(active))
+        if active:
+            tickets = [t for t in tickets if t.is_active()]
+        if tickets:
+            serializer = IrodsAccessTicketSerializer(tickets, many=True)
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            return Response(
+                {'detail': IRODS_TICKETS_NOT_FOUND_MSG},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+
+class IrodsAccessTicketRetrieveAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIBaseProjectMixin, APIView
+):
+    """
+    Retrieve an iRODS access ticket for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/retrieve/{IrodsAccessTicket.sodar_uuid}``
+
+    **Methods:** ``GET``
+
+    **Returns:**
+
+    - ``ticket``: iRODS access ticket
+    """
+
+    http_method_names = ['get']
+    permission_required = 'samplesheets.edit_sheet'
+
+    def get(self, request, *args, **kwargs):
+        """GET request for retrieving an iRODS access ticket"""
+        try:
+            ticket = IrodsAccessTicket.objects.get(
+                sodar_uuid=self.kwargs.get('irodsaccessticket')
+            )
+        except IrodsAccessTicket.DoesNotExist:
+            return Response(
+                {'detail': IRODS_TICKETS_NOT_FOUND_MSG},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+        serializer = IrodsAccessTicketSerializer(ticket)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class IrodsAccessTicketCreateAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIBaseProjectMixin, CreateAPIView
+):
+    """
+    Create an iRODS access ticket for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/create/{Project.sodar_uuid}``
+
+    **Methods:** ``POST``
+
+    **Parameters:**
+
+    - ``assay``: Assay primary key
+    - ``path``: iRODS path
+    - ``label``: Label (optional)
+    - ``date_expires``: Expiration date (YYYY-MM-DDThh:mm:ssZ, optional)
+
+    **Returns:**
+
+    - ``ticket``: iRODS access ticket
+    """
+
+    permission_required = 'samplesheets.edit_sheet'
+
+    def post(self, request, *args, **kwargs):
+        """POST request for creating an iRODS access ticket"""
+        irods_backend = get_backend_api('omics_irods')
+        self.project = self.get_project()
+        serializer = IrodsAccessTicketSerializer(
+            data=request.data, partial=True
+        )
+        serializer.context['project'] = self.project
+        serializer.context['user'] = self.request.user
```

Comment:
These should be set in `serializer.create()`, see `IrodsDataRequestSerializer`

Reply:
As this data is needed for validation, I've added similar method `get_serializer_context`. Let me know, if I should edit something.

Reply:
This seems fine. Altough I believe `project` should be added automatically by the SODAR Core base API classes, so it should already be there after you call `super().get_serializer_context()`? Please verify and omit the project re-retrieval if already there.